### PR TITLE
clean & refactor

### PR DESCRIPTION
**Pull request by ngoldman**
Got rid of some unnecessary things, cleaned up code, improved package.json, lots of housekeeping basically. Should be more straightforward and have more parity with express examples and conventions than before.

@chelm I'm unable to get previews working locally at this point. Getting the following errors with what's provided in the default `index.ejs`

with `http://localhost:1337/gist/6178185/preview`

```
ReferenceError: /Users/nate/dev/github/koop-sample-app/node_modules/koop-gist/views/demo.ejs:65
   63|         $('#'+mapDom).css({ width:document.width, height: document.height });
   64|         var koop = koopMap( mapDom );
>> 65|         koop.add(location.origin + '/gist/' + '<%= id %>' + '/FeatureServer/0');
   66|       });
   67| 
   68|     </script>

id is not defined
   at eval (eval at <anonymous> (/Users/nate/dev/github/koop-sample-app/node_modules/ejs/lib/ejs.js:455:12), <anonymous>:9:28)
   at /Users/nate/dev/github/koop-sample-app/node_modules/ejs/lib/ejs.js:482:14
   at View.exports.renderFile [as engine] (/Users/nate/dev/github/koop-sample-app/node_modules/ejs/lib/ejs.js:348:31)
   at View.render (/Users/nate/dev/github/koop-sample-app/node_modules/express/lib/view.js:93:8)
   at EventEmitter.app.render (/Users/nate/dev/github/koop-sample-app/node_modules/express/lib/application.js:566:10)
   at ServerResponse.res.render (/Users/nate/dev/github/koop-sample-app/node_modules/express/lib/response.js:938:7)
   at controller.preview (/Users/nate/dev/github/koop-sample-app/node_modules/koop-gist/controller/index.js:108:9)
   at Layer.handle [as handle_request] (/Users/nate/dev/github/koop-sample-app/node_modules/express/lib/router/layer.js:82:5)
   at next (/Users/nate/dev/github/koop-sample-app/node_modules/express/lib/router/route.js:110:13)
   at Route.dispatch (/Users/nate/dev/github/koop-sample-app/node_modules/express/lib/router/route.js:91:3)
```

with `http://localhost:1337/github/chelm/grunt-geo/forks/preview`

```
ReferenceError: /Users/nate/dev/github/koop-sample-app/node_modules/koop-github/views/demo.ejs:63
   61|         $('#'+mapDom).css({ width:document.width, height: document.height });
   62|         var koop = koopMap( mapDom );
>> 63|         koop.add(location.origin + '/github/<%= user %>/<%= repo %>/<%= file %>/FeatureServer/0');
   64|       });
   65|     </script>
   66|   </body>

user is not defined
   at eval (eval at <anonymous> (/Users/nate/dev/github/koop-sample-app/node_modules/ejs/lib/ejs.js:455:12), <anonymous>:9:28)
   at /Users/nate/dev/github/koop-sample-app/node_modules/ejs/lib/ejs.js:482:14
   at View.exports.renderFile [as engine] (/Users/nate/dev/github/koop-sample-app/node_modules/ejs/lib/ejs.js:348:31)
   at View.render (/Users/nate/dev/github/koop-sample-app/node_modules/express/lib/view.js:93:8)
   at EventEmitter.app.render (/Users/nate/dev/github/koop-sample-app/node_modules/express/lib/application.js:566:10)
   at ServerResponse.res.render (/Users/nate/dev/github/koop-sample-app/node_modules/express/lib/response.js:938:7)
   at controller.preview (/Users/nate/dev/github/koop-sample-app/node_modules/koop-github/controller/index.js:195:10)
   at Layer.handle [as handle_request] (/Users/nate/dev/github/koop-sample-app/node_modules/express/lib/router/layer.js:82:5)
   at next (/Users/nate/dev/github/koop-sample-app/node_modules/express/lib/router/route.js:110:13)
   at Route.dispatch (/Users/nate/dev/github/koop-sample-app/node_modules/express/lib/router/route.js:91:3)
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,41 @@
...
-node server.js
-```
-5. you can try fetching a resource directly in the browser to confirm koop is running
-[http://localhost:1337/github/benbalter/dc-wifi-social/bars/](http://localhost:1337/github/benbalter/dc-wifi-social/bars/)
-
-##Additional documentation
+1. Clone this repository on your machine.
+
+  ```
+  git clone git@github.com:ngoldman/koop-sample-app.git
```
Comment (jgravois) [inline README.md]: ~~ngoldman~~ / koopjs

**Comment by jgravois** [comment]
nice work @ngoldman!

**Comment by chelm** [comment]
@ngoldman Looking good. The link in the readme is the only thing I see. 

The issue with the preview not working seems to be in the relationship between the code in the providers and the version of `ejs` in the sample app. I'm going to look at it more closely. It'd be great to just do away with the preview routes in each provider and come up with boiler plate, centralized route/preview

**Comment by chelm** [comment]
Okay yes so for latest version of the ejs module we need to do away with passing "locals" to template like so: https://github.com/koopjs/koop-github/blob/master/controller/index.js#L195

and just pass the `req.params` instead...I cant open some PRs to cover the providers in the sample app starting with github and gist

**Comment by chelm** [comment]
what's really annoying is that the ejs in "koop" is being used over the ejs in providers. I think a nice way to solve this would be to have providers use their own template module, whatever that may be. 

So for the sake of this issue, the koop module itself can use ejs v.2* and github could use ejs v.1* and render precompiled templates like so: 

```javascript

res.send(ejs.render(fs.readFileSync(__dirname + '/../views/demo.ejs').toString(), { locals: req.params }));

```

File: README.md (hunk 1 above)
Comment (ngoldman) [inline README.md]: @jgravois fixed! thanks for catching that

**Comment by chelm** [comment]
Merging, and we can open an issue about the proview routes

**Comment by ngoldman** [comment]
sounds like it might make sense to migrate to having preview as another plugin, e.g.

```
var preview = require('koop-preview')
koop.register(preview)
```

And maybe have a `preview` route above providers, like

```
/preview/github/benbalter/dc-wifi-social/bars/
```

The reasoning being koop and providers don't really need templates and shouldn't have to know how to render things for a web page, they should just do the job of ETL for GeoJSON / Feature Service API work. A `koop-preview` module could encapsulate all preview and template logic (as well as esri-leaflet etc) and keep it out of everthing else

**Comment by ngoldman** [comment]
@chelm thanks, sounds good, see above for a little idea ^

**Comment by chelm** [comment]
@ngoldman I can dig it, sounds like a good idea

**Pull request merged**